### DISCUSSION TITLE
Can't custom login.

### DISCUSSION THREAD

**Discussion by athevier** [Structed/godot-playfab#123] [Q&A]
Hi,

First, thank you for providing a plugin allowing us to use Playfab in Godot.

I have experience with playfab in Unreal in C++ and in Unity also but I can't manage to use the plugin for Godot despite reading the examples, especially since there are things different than what I'm used to.

So for instance, in Unreal, if I want to login with a custom ID I would do this : 

`
void APlayfabManager::PlayfabAuthentification()
{
	
    clientAPI = IPlayFabModuleInterface::Get().GetClientAPI();

	PlayFab::ClientModels::FLoginWithCustomIDRequest request;

  	request.CustomId =TEXT("UNREAL DEVICE");

	request.CreateAccount = true;

	clientAPI->LoginWithCustomID(request,
		PlayFab::UPlayFabClientAPI::FLoginWithCustomIDDelegate::CreateUObject(this, &APlayfabManager::OnPlayfabAuthentificationSucceeded),
		PlayFab::FPlayFabErrorDelegate::CreateUObject(this, &APlayfabManager::OnPlayfabAuthentificationFailed)
	);
}

void APlayfabManager::OnPlayfabAuthentificationSucceeded(const PlayFab::ClientModels::FLoginResult& Result)
{
	UE_LOG(LogTemp, Log, TEXT("Congratulations, you made your first successful API call!"));

}

void APlayfabManager::OnPlayfabAuthentificationFailed(const PlayFab::FPlayFabCppError& ErrorResult)
{
	UE_LOG(LogTemp, Error, TEXT("Something went wrong with your first API call.\nHere's some debug information:\n%s"), *ErrorResult.GenerateErrorReport());

}
`

Which is the authentification function with the two functions that handle the callbacks in case of login successful and login error.

Now in your examples in the project, I believe that this is the custom id login : 

`func _on_AnonLogin_pressed():
	$Login.hide()
	_show_progess()
	var combined_info_request_params = GetPlayerCombinedInfoRequestParams.new()
	combined_info_request_params.show_all()
	var player_profile_view_constraints = PlayerProfileViewConstraints.new()
	combined_info_request_params.ProfileConstraints = player_profile_view_constraints

	if PlayFabManager.client_config.login_type == PlayFabClientConfig.LoginType.LOGIN_CUSTOM_ID:
		PlayFabManager.client.login_with_custom_id(PlayFabManager.client_config.login_id, false, combined_info_request_params)
	else:
		PlayFabManager.client.login_with_custom_id(UUID.v4(), true, combined_info_request_params)`
		
		
What I don't get is where are the functions that handle the login success and the login failure ? 

And I tried to reproduce a similar function like this called in _ready function of a node : 

`
       var combined_info_request_params = GetPlayerCombinedInfoRequestParams.new()
	combined_info_request_params.show_all()
	PlayFabManager.client.login_with_custom_id(PlayFabManager.client_config.login_id, false, combined_info_request_params)

	
`

But it just does not login and I have no error message.

What exactly am I doing wrong here ?

Thanks.

**Comment by Structed** [comment]
Hey! Sorry for the long wait - I need to do a better job of checking my notifications!

The plugin works completely different to all other SDKs, as it's hand-crafted and looks to cater to developer needs in the Godot world. So forget about a lot of how you use the other SDKs 😉. What still is good knowledge, is how the PlayFab systems and it's general Client/Server/Admin APIs behave!


In order to react to a login/login failure, you need to connect Signals! See the [Connecting SIgnals](https://github.com/Structed/godot-playfab/blob/develop/docs/user/connecting-signals.md) page in the docs for this!
